src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+
+While experimenting with service workers, the Google Search team made sure to
+keep the experiments running across a number of scheduled backend updates, to
+ensure that the metrics and user experience would more closely match what return
+users would end up seeing in the real-world.
+
+{% Aside 'caution' %}
+It's important to remember that shipping a service worker
+is **not a one-time deployment**—you need to have a process in place, tailored to
+your own production infrastructure, to make sure that updates happen smoothly
+over time!
+{% endAside %}
+
+**Solution**
+
+After testing a number of different configuration options, the Google Search
+team found that the following setup provided the right balance between freshness
+and cache-utilization.
+
+The service worker script URL is served with the
+`Cache-Control: private, max-age=1500` (1500 seconds, or 25 minutes) response
+header, and it's
+[registered with updateViaCache set to 'all'](https://developers.google.com/web/updates/2018/06/fresher-sw#updateviacache)
+to ensure that the header is honored. The Google Search web backend is, as you
+might imagine, a large, globally distributed set of servers that requires as
+close to 100% uptime as possible. Deploying a change that would affect the
+service worker script's contents is done in a rolling fashion. If a user hits a
+backend that has been updated, and then quickly navigates to another page which
+hits a backend that hasn't yet received the updated service worker, they'd end
+up flip-flopping between versions multiple times. Therefore, telling the browser
+to only bother checking for an updated script if 25 minutes has passed since the
+last check does not have a significant downside. The upside of opting-in to this
+behavior is cutting down significantly on the traffic received by the endpoint
+that dynamically generates the service worker script. Additionally, an ETag
+header is set on the service worker script's HTTP response, ensuring that when
+an update check is made after 25 minutes has passed, the server can respond
+efficiently with an [HTTP 304](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304)
+response if there haven't been any updates to the service worker deployed in the
+interim.
+
+While some interactions within the Google Search web app do use single page
+app-style navigations (i.e. via the
+[History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API#Adding_and_modifying_history_entries)),
+for the most part, Google Search is a traditional web app that uses "real"
+navigations. This comes into play when the team decided that it would be
+effective to use two options that accelerate the service worker update
+lifecycle:
+[`clients.claim()`](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#clientsclaim)
+and
+[`skipWaiting()`](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#skip_the_waiting_phase).
```

Comment:
Remove the 'will' usage:

Clicking around Google Search's interface generally ends up navigating to new HTML documents. Calling `skipWaiting` ensures that an updated service worker gets a chance to handle those ... Similarly, calling ... means that the updated service worker gets a chance to ...